dataline-integration-tests/src/test/java/io/dataline/integration_tests/destinations/TestBigQueryDestination.java

```diff
@@ -0,0 +1,184 @@
+import com.google.cloud.bigquery.TableResult;
+import com.google.common.base.Charsets;
+import io.dataline.commons.json.Jsons;
+import io.dataline.workers.WorkerUtils;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.StreamSupport;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.toList;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class TestBigQueryDestination extends BaseIntegrationTestCase {
+
+  private static final BigQuery BQ = BigQueryOptions.getDefaultInstance().getService();
+
+  private String datasetName;
+  private Dataset dataset;
+  private Process process;
+
+  @BeforeEach
+  public void setUpBigQuery() throws IOException {
+    datasetName = "dataline_tests_" + RandomStringUtils.randomAlphanumeric(8);
+    DatasetInfo datasetInfo = DatasetInfo.newBuilder(datasetName).build();
+    dataset = BQ.create(datasetInfo);
+    System.out.println("BQ Dataset " + datasetName + " created...");
+
+    writeConfigFileToJobRoot();
+    process = startTarget();
+  }
+
+  @AfterEach
+  public void tearDownBigQuery() {
+    WorkerUtils.closeProcess(process);
+
+    // allows deletion of a dataset that has contents
+    BigQuery.DatasetDeleteOption option = BigQuery.DatasetDeleteOption.deleteContents();
+
+    boolean success = BQ.delete(dataset.getDatasetId(), option);
+    if (success) {
+      System.out.println("BQ Dataset " + datasetName + " deleted...");
+    } else {
+      System.out.println("BQ Dataset cleanup for " + datasetName + " failed!");
+    }
+  }
+
+  @Test
+  public void runTest() throws IOException, InterruptedException {
+    writeResourceToStdIn("singer-tap-output.txt", process);
+
+    await()
+        .atMost(10, SECONDS)
+        .until(
+            () -> {
+              try {
+                return getExchangeRateTable().getTotalRows() == 2L;
+              } catch (Exception e) {
+                return false;
+              }
+            });
+
+    TableResult results = getExchangeRateTable();
+    List<FieldValueList> actualList =
+        StreamSupport.stream(results.iterateAll().spliterator(), false).collect(toList());
+
+    List<FieldValueList> expectedList =
+        Arrays.asList(
+            FieldValueList.of(
+                Arrays.asList(fieldValue(1598918400.0), fieldValue("0.0"), fieldValue("2.0"))),
+            FieldValueList.of(
+                Arrays.asList(fieldValue(1598918400.0), fieldValue("1.0"), fieldValue("7.0"))));
+
```

Comment:
also assert the list sizes are equal?

Reply:
the new method of testing (converting to a csv table, basically) does this implicitly